pkg/process/checks/process_data.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/process/metadata"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcData is a singleton ProcessData.
+var ProcData = NewProcessData()
```

Comment:
This is only used by the connections check. Any reason to have it be a global/singleton?

Reply:
We want to eventually use this as the global source of process data which would also provide data to the processes check. This way we only traverse once `proc` once per check cycle.

Reply:
Could we leave this global out until that is true? Right now it would allocate if you import this package.

There is also some danger here with a single source of processes. Right now the docker-proxy filtering gets an up-to-date list of processes. Using stale data from the last check could impact the accuracy.

Reply:
I've removed the singleton for now. 

Ideally we would sync all updates of process data with their respective checks. I'm not sure if I'm missing something, but it looks like there already a possibility of data inaccuracy as the connections check and the process check happen serially. So this is a best effort.